### DISCUSSION TITLE
How to check if the `create_or_update_secret` method succeeded?

### DISCUSSION THREAD

**Discussion by MarkRushB** [hvac/hvac#967] [Q&A]
link: https://hvac.readthedocs.io/en/stable/source/hvac_api_secrets_engines.html#hvac.api.secrets_engines.KvV2.create_or_update_secret

From the doc I cannot get more detailed info, I found there is no status_code returned by `create_or_update_secret` method, and I am wondering if this method failed, what would be response like?

**Comment by briantist** [comment]
Hi @MarkRushB !

The `create_or_update_secret` method returns the raw `requests.Response` object from the [Vault HTTP API call](https://developer.hashicorp.com/vault/api-docs/secret/kv/kv-v2#create-update-secret).

Although you could get a status code from that response, in practice it should always be a `200` because if the method failed, an exception would be raised from one of the `hvac.exceptions` classes.

Our method calls the adapter's `post` method
https://hvac.readthedocs.io/en/stable/_modules/hvac/api/secrets_engines/kv_v2.html#KvV2.create_or_update_secret

which in turn calls the adapter's `request` method:
https://hvac.readthedocs.io/en/stable/_modules/hvac/adapters.html#RawAdapter.request

And that method uses the adapter's `_raise_for_error` method:
https://hvac.readthedocs.io/en/stable/_modules/hvac/adapters.html#RawAdapter

Which eventually calls the `VaultError.from_status` classmethod to produce a specific exception (here you can see the mapping of status code to exception class):
https://hvac.readthedocs.io/en/stable/_modules/hvac/exceptions.html#VaultError.from_status

---

As of `hvac==1.1.0`, the exceptions have additional helpful fields in them that can help you handle errors better:
*  `errors`
* `method`
* `url`
* `test`
* `json`
